### ISSUE TITLE
Unable to install in Legacy BIOS.

### ISSUE THREAD

**Issue by gilbertoferreira** [linuxdeepin/developer-center#3265]
### SUMMARY | 问题概要

Tried to make a fresh installation in a laptop ACER Aspire A515-41G-13U1 with legacy bios.
After 40% installation procedure, got error about grub efi etc, and the installation aborted.

### STEPS TO REPRODUCE | 复现步骤

1 - download iso
2 - burn a cd or a usb stick
3 - boot laptop and initialize from usb boot
4 - choose the language, keybord layout and create partitions
5 - installation begins and then, after 40% crash about grub error

### OBSERVED RESULT | 观察到的结果

Grub crashes during installation
### EXPECTED RESULT | 期望的结果

I expected installation finish sucessful, No luck!

**Comment by BLumia** [comment]
> got error about grub efi etc

Could you provided the actual error displayed on screen? It sounds like you are probably trying to install via EFI instead of Legacy MBR method.

**Comment by gilbertoferreira** [comment]
https://user-images.githubusercontent.com/10462980/176192796-7a2c6e84-da97-4823-9bb3-e5ce56c19530.mp4

**Comment by gilbertoferreira** [comment]
![IMG_20220628_101547](https://user-images.githubusercontent.com/10462980/176192918-43ed22c9-6472-443b-95ff-31923d8d8860.jpg)

**Comment by BLumia** [comment]
![image](https://user-images.githubusercontent.com/10095765/176332903-49486aa6-109a-4f94-b858-1a0c48bfd78d.png)

Seems you are using UEFI installation method instead of Legacy/MBR method, please consider double checking your BIOS settings and ensure you only enabled Legacy/MBR method. If you decide to use the EFI method, please create an EFI partition manually if you are not using the full disk installation method since EFI installation requires an EFI partition to work, or you can use the full disk installation instead and it will create an EFI partition for you.

**Comment by gilbertoferreira** [comment]
> ![image](https://user-images.githubusercontent.com/10095765/176332903-49486aa6-109a-4f94-b858-1a0c48bfd78d.png)
> 
> Seems you are using UEFI installation method instead of Legacy/MBR method, please consider double checking your BIOS settings and ensure you only enabled Legacy/MBR method. If you decide to use the EFI method, please create an EFI partition manually if you are not using the full disk installation method since EFI installation requires an EFI partition to work, or you can use the full disk installation instead and it will create an EFI partition for you.

I am not using UEFI at all! 
Every Linux system I install in this laptop always used Legacy BIOS.
I had triple check the BIOS and always legacy!
Is there any options in boot time that I can pass to the installer as a parameter??
Thanks

**Comment by BLumia** [comment]
We actually have legacy-only boot test-case covered and we cannot reproduce this issue now. Could you provide a screenshot/photo of your BIOS boot setting? Also, could you try it if it works using EFI only method? Thanks!

**Comment by BLumia** [comment]
Congratulations!

> I just now tried to get used myself with the grand search which for some reason active everytime I am typing and press space bar.

FYI you can change the default shortcut of the grand search from the control center (settings) app :)

**Comment by gilbertoferreira** [comment]
Hi! I try again and no luck.
Why I can not install DeepIn without UEFI??
I can do that with Ubuntu, Kubuntu, Debian, VoidLinux, Devuan, Alpine Linux and many others distros!
Why DeepIn installer is so hard to detect that I don't want to use UEFI boot????
Why so difficult????

**Comment by BLumia** [comment]
We are pretty sure installing without UEFI is possible, so you need to provide details (that ensure someone other than you can reproduce this issue) so we could look into it.

If the reproduce step and the result is very same as the original issue, please do provide the information that https://github.com/linuxdeepin/developer-center/issues/3265#issuecomment-1170648108 asked for, if not, please do provide more details.

**Comment by gilbertoferreira** [comment]
Boot normal in legacy mode

![IMG_20220726_100456](https://user-images.githubusercontent.com/10462980/181016281-695f36d3-b8c2-4eb6-a309-99c25c6c7410.jpg)

Set language ok!
![IMG_20220726_100712](https://user-images.githubusercontent.com/10462980/181016409-3bc12a28-d123-45be-a55e-ff90f6250a43.jpg)

Create partitions ok!

![IMG_20220726_100727](https://user-images.githubusercontent.com/10462980/181016501-62c17d52-45e6-468a-9605-3ca5b1141eac.jpg)

After a while during the installation

![IMG_20220726_101150](https://user-images.githubusercontent.com/10462980/181016637-79c32fbf-aebc-4738-b78e-0d33d8d61b55.jpg)

![IMG_20220726_101202](https://user-images.githubusercontent.com/10462980/181016767-1b3bba75-be39-413b-863a-20b2b5917a7a.jpg)

**Comment by BLumia** [comment]
The photo we asked for (BIOS setting screen with Legacy selected) is still not provided. We still cannot confirm this as an issue...

**Comment by gilbertoferreira** [comment]
> The photo we asked for (BIOS setting screen with Legacy selected) is still not provided. We still cannot confirm this as an issue...

Here!
![IMG_20221102_102157_379](https://user-images.githubusercontent.com/10462980/199501234-8574b200-c0e3-47d6-95df-7b958ec7d899.jpg)

**Comment by gilbertoferreira** [comment]
I also tried again, but this time with the 23 preview and still nothing at all.
![IMG_20221102_084534_464](https://user-images.githubusercontent.com/10462980/199501605-b754875e-dc28-433f-b6bd-0c68ca062025.jpg)
![IMG_20221102_084502_285](https://user-images.githubusercontent.com/10462980/199501614-a141424e-affa-4d50-bb64-66c18ae93ebf.jpg)

**Comment by BLumia** [comment]
btw, which software/method is you used to write ISO to image? e.g. `dd` / `rufus` / other possible application/command.

**Comment by gilbertoferreira** [comment]
> btw, which software/method is you used to write ISO to image? e.g. `dd` / `rufus` / other possible application/command.

I am not using either dd or rufus. Altough I alread tried used a pendriver which I wrote with balena etcher.
Usually I have an old Android phone (rooted) with DriveDroid installed.
This same method I used to install Ubuntu, Debian, Kubuntu, VoidLinux, LinuxMint, Devuan, ZorinOS and a tons of others distros, like Fedora, CentOS, UbuntuDDE Remix, etc etc etc....
Dont know why Deepin Linux doesn't work,

**Comment by babyfengfjx** [comment]
Try adding an EFI partition here：
![image](https://user-images.githubusercontent.com/24642999/199939128-cac66263-bfd4-4991-9dbc-f8c6cdae9299.png)

**Comment by gilbertoferreira** [comment]
Ok! It's work.
But after a successful installation, I am again stuck with the message bellow:

No bootable device -- insert boot disk and press any key

So I booted my laptop with LinuxMint and do chroot.
Then I needed to install grub-pc and only then I could boot the system properly.
This is embarrassing to say the least.

![IMG_20221104_230519_633](https://user-images.githubusercontent.com/10462980/200121175-153fa1a3-5cda-469a-a589-5a7717ae6e12.jpg)
